src/sagas/RootSaga.js

```diff
@@ -1,138 +1,110 @@
-  saveUserProfileReset,
+  FETCH_PROFILE,
+  fetchProfileBegin,
+  fetchProfileSuccess,
+  fetchProfileFailure,
+  fetchProfileReset,
+  fetchProfile as fetchProfileAction,
+  SAVE_PROFILE,
+  saveProfileBegin,
+  saveProfileSuccess,
+  saveProfileFailure,
+  saveProfileReset,
   closeEditableField,
-  SAVE_USER_PROFILE_PHOTO,
-  saveUserProfilePhotoBegin,
-  saveUserProfilePhotoSuccess,
-  saveUserProfilePhotoFailure,
-  saveUserProfilePhotoReset,
-  DELETE_USER_PROFILE_PHOTO,
-  deleteUserProfilePhotoBegin,
-  deleteUserProfilePhotoSuccess,
-  deleteUserProfilePhotoFailure,
-  deleteUserProfilePhotoReset,
+  SAVE_PROFILE_PHOTO,
+  saveProfilePhotoBegin,
+  saveProfilePhotoSuccess,
+  saveProfilePhotoFailure,
+  saveProfilePhotoReset,
+  DELETE_PROFILE_PHOTO,
+  deleteProfilePhotoBegin,
+  deleteProfilePhotoSuccess,
+  deleteProfilePhotoFailure,
+  deleteProfilePhotoReset,
 } from '../actions/profile';
 
-let userAccountApiService = null;
-
-const PROP_TO_STATE_MAP = {
-  fullName: 'name',
-  userLocation: 'country',
-  education: 'levelOfEducation',
-  socialLinks: socialLinks => socialLinks.filter(({ socialLink }) => socialLink !== null),
-};
-
-export const mapDataForRequest = (props) => {
-  const state = {};
-  Object.keys(props).forEach((prop) => {
-    const propModifier = PROP_TO_STATE_MAP[prop] || prop;
-    if (typeof propModifier === 'function') {
-      state[prop] = propModifier(props[prop]);
-    } else {
-      state[propModifier] = props[prop];
-    }
-  });
-  return state;
-};
-
-
-export function* saveUserProfile(action) {
-  const { username, userAccountState } = action.payload;
+import * as ProfileApiService from '../services/ProfileApiService';
+
+export function* handleFetchProfile(action) {
+  const { username } = action.payload;
 
   try {
-    yield put(saveUserProfileBegin());
+    yield put(fetchProfileBegin());
+    const profile = yield call(
+      ProfileApiService.getProfile,
+      username,
+    );
 
-    // Use call([context, fnName], ...args) for proper context on apiService
-    const userAccount = yield call(
-      [userAccountApiService, 'saveUserAccount'],
+    yield put(fetchProfileSuccess(profile));
+    yield put(fetchProfileReset());
+  } catch (e) {
+    yield put(fetchProfileFailure(e.message));
+  }
+}
+
+export function* handleSaveProfile(action) {
+  const { username, userAccountState } = action.payload;
+  try {
+    yield put(saveProfileBegin());
+    const profile = yield call(
+      ProfileApiService.patchProfile,
       username,
-      mapDataForRequest(userAccountState),
+      userAccountState,
     );
 
     // Tells the profile form that
-    yield put(saveUserProfileSuccess());
-    // TODO: export the fetchUserAccountSuccess action from frontend-auth so we can
-    // dry this up.
-    yield put({
-      type: 'FETCH_USER_ACCOUNT_SUCCESS',
-      payload: { userAccount },
-    });
+    yield put(saveProfileSuccess());
+    yield put(fetchProfileSuccess(profile));
```

Comment:
This is no longer a load of the user account - it's a load of the profile, so we can remove the dependency here on anything from frontend-auth, which feels nice.